grails-app/services/org/pih/warehouse/data/ProductSupplierPreferenceDataService.groovy

```diff
@@ -24,13 +24,14 @@ class ProductSupplierPreferenceDataService {
 
     Boolean validate(ImportDataCommand command) {
         log.info "Validate data " + command.filename
+        String dateFormat = "MM/dd/yyyy"
         command.data.eachWithIndex { params, index ->
 
-            def supplierCode = params.supplierCode
-            def organizationCode = params.organizationCode
-            def preferenceTypeName = params.preferenceTypeName
-            def validityStartDate = params.validityStartDate
-            def validityEndDate = params.validityEndDate
+            String supplierCode = params.supplierCode
+            String organizationCode = params.organizationCode
+            String preferenceTypeName = params.preferenceTypeName
+            params.validityStartDate = params?.validityStartDate?.format(dateFormat)
```

Comment:
I'm confused why you keep trying to do this. You were right about using DateJavaType as it's difficult to convert from LocalDate to Date. But once you define those columns as DataJavaType you don't need to do anything else with the date parameters (no parsing, no formatting). Just use the `validatityStartDate` and `validatityEndDate` as Date objects.

So the PR should look like this.
https://github.com/openboxes/openboxes/pull/4228/files

Reply:
@jmiranda I pushed the code that you proposed. I tried to parse the date because with the current version of the code it looks like that:
![image](https://github.com/openboxes/openboxes/assets/83239466/817bb6b6-e079-4689-86f4-0a27f74cb2a1)
I think it's not the date format that we would like to have.